apps/epic-web/src/server/lesson-for-device-req.ts

```diff
@@ -27,11 +28,16 @@ export async function lessonForDeviceReq({
       ...(user && {user}),
       useSolution: isSolution,
     })
-
     if (lessonForDevice) {
       res.status(200).json(lessonForDevice)
     } else {
-      res.status(404).end()
+      if (user) {
+        // unauthorized
+        res.status(403).json(lessonForDevice)
```

Comment:
```suggestion
        res.status(403).end()
```

in the else the lessonForDevice will be `undefined`